### PR TITLE
Use `lootGameName()` and `displayGameName()` in initial places

### PR DESCRIPTION
**Pull request by Twinki14**
# Motivations
- https://github.com/ModOrganizer2/modorganizer-game_ttw/pull/36#discussion_r1605285526 Surfaced desires to change the Fallout TTW plugin's gameName to something more obvious, from TTW to Tale of Two Wastelands, but that would cause existing profiles to break, `displayGameName()` added in https://github.com/ModOrganizer2/modorganizer-uibase/pull/141 gives us the option to customize the way the game plugin is displayed without affecting the integral `gameName()`
- Similarly https://github.com/ModOrganizer2/modorganizer-game_ttw/pull/36 attempted to fix the sort button LOOT cli argument by changing the `gameShortName()` as 'TTW' isn't a valid option for LOOT, but that also might break existing profiles in some way, `lootGameName()` will let us fix this without the possibility of breaking existing profiles

# Modifications
- Use `displayGameName()` in create instance dialogs & the main window, this doesn't cover all places `gameName()` was being used for purely display reasons, but it covers the bulk
- Use `lootGameName()` instead of `gameShortName()` for LOOT cli initiation

### CODE REVIEW

**Comment by Twinki14** [comment]
Note, it's expected the build will fail until https://github.com/ModOrganizer2/modorganizer-uibase/pull/141 is merged

File: src/createinstancedialogpages.cpp [hunk 1]
```diff
@@ -570,7 +571,8 @@ void GamePage::fillList()
       continue;
     }
 
-    if (!m_filter.matches(g->game->gameName())) {
+    if (!m_filter.matches(g->game->gameName()) &&
```
Comment (Holt59) [inline src/createinstancedialogpages.cpp:574]: I'm not sure this should be kept - The `gameName()` is now internal only, so using it to filter for users could lead to unexpected behavior.

Comment (Twinki14) [reply src/createinstancedialogpages.cpp:574]: This area is just the filter search when creating a profile, I don't think it'd hurt anything to continue using the gameName alongside the displayName

For tale of two wastelands for example I can search "TTW" or "Tale of Two wastelands"

**Comment by Holt59** [comment]
Merged manually due to conflict on translation file (+ added change to status bar), https://github.com/ModOrganizer2/modorganizer/commit/1871f32a8db7e707a6d96a41f09147d8558e0003

**Pull request closed without merging**